tests/utils/k8s.go

```diff
@@ -66,9 +69,64 @@ func DropNamespace(clientSet *kubernetes.Clientset, name string) error {
 	})
 }
 
+func getPodLog(clientSet *kubernetes.Clientset, namespace string, pod string, lines int64) (string, error) {
+	opts := v1.PodLogOptions{}
+	if lines > 0 {
+		opts.TailLines = &lines
+	}
+	req := clientSet.CoreV1().Pods(namespace).GetLogs(pod, &opts)
+
+	logSteam, err := req.Stream()
+	if err != nil {
+		return "", err
+	}
+	defer logSteam.Close()
+
+	logBuffer := new(bytes.Buffer)
+	_, err = io.Copy(logBuffer, logSteam)
+	if err != nil {
+		return "", err
+	}
+
+	return logBuffer.String(), nil
+}
+
+func podLogContains(clientSet *kubernetes.Clientset, namespace string, pod string, text string) (bool, error) {
```

Comment:
is it an optimization to aviod getting full log from a pod? what would be the drawback of `return strings.Contains(getPodLog(<parameters>), text)`?

Reply:
This is an optimization, as I have no idea how big the logs could be. RAM consumption is the only drawback I think.

Reply:
thanks for addressing that!